cmd/metal-api/internal/datastore/network.go

```diff
@@ -91,42 +92,48 @@ func (p *NetworkSearchQuery) generateTerm(rs *RethinkStore) *r.Term {
 	}
 
 	for _, prefix := range p.Prefixes {
-		ip, length := utils.SplitCIDR(prefix)
+		pfx, err := netip.ParsePrefix(prefix)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse prefix %w", err)
+		}
+		ip := pfx.Addr()
+		length := pfx.Bits()
 
 		q = q.Filter(func(row r.Term) r.Term {
 			return row.Field("prefixes").Map(func(p r.Term) r.Term {
 				return p.Field("ip")
-			}).Contains(r.Expr(ip))
+			}).Contains(r.Expr(ip.String()))
```

Comment:
I think in a follow-up PR we should write an integration test (similar to how it was done for machine entity) to verify that all these filters are really working. I can do this after this was merged.

Reply:
We had one, but struggled to make this work in the integration test in CI

Reply:
https://github.com/metal-stack/metal-api/pull/541